src/main/kotlin/me/lensvol/blackconnect/settings/BlackConnectGlobalSettings.kt

```diff
@@ -12,12 +12,24 @@ import com.intellij.util.xmlb.annotations.Attribute
 class BlackConnectGlobalSettings : PersistentStateComponent<BlackConnectGlobalSettings> {
 
     companion object {
-        fun getInstance(): BlackConnectGlobalSettings = service()
+        fun getInstance() = service<BlackConnectGlobalSettings>()
     }
 
+    @Attribute
+    var spawnBlackdOnStartup: Boolean = true
+
     @Attribute
     var showSaveTriggerOptIn: Boolean = true
 
+    @Attribute
+    var bindOnHostname: String = "localhost"
+
+    @Attribute
+    var bindOnPort: Int = 45484
+
+    @Attribute
+    var blackdBinaryPath: String = "/Users/lensvol/.local/bin/blackd"
```

Comment:
The blackd path is likely to be machine-specific, would it be better to leave it as an empty string by default and auto-detect the actual path if this setting is empty? It's better not to save the auto-detected path to settings similar to how it's done in IntelliJ with the Git path.

For auto-detect in `PATH` see `PathEnvironmentVariableUtil.findInPath()`. You can access it synchronously. If you want to pre-activate virtualenv in order to source `PATH` from there, then it might involve running an external process, therefore it should be done in a non-EDT thread.

Reply:
Oops, my bad! It should have been an empty string by default.

Reply:
I am not really sure how to auto-detect yet, maybe it makes more sense to ship without it at the moment?

Reply:
I would start with a simplistic approach I've described above: just auto-detect `blackd` in user's `PATH` without virtualenv. See `PathEnvironmentVariableUtil.findInPath()` I've mentioned above. Then we could improve upon that by looking in `PATH` of your virtualenv.